lib/fizzy/execute.hpp

```diff
@@ -2,30 +2,37 @@
 
 #include "types.hpp"
 #include <cstdint>
+#include <functional>
 
 namespace fizzy
 {
+// The result of an execution.
+struct execution_result
+{
+    // true if execution resulted in a trap
+    bool trapped;
+    // the resulting stack (e.g. return values)
+    // NOTE: this can be either 0 or 1 items
+    std::vector<uint64_t> stack;
+};
+
+struct Instance;
+
+using ImportedFunction = execution_result (*)(Instance&, const std::vector<uint64_t>&);
```

Comment:
Added `Instance` here, but I think there needs to be a safer way, passing only exported memory/globals/tables. `Instance` has all of them, not only exported.
Maybe not for this PR...